### PR TITLE
fix(rest-api): improve openapi-apis by adding required on discriminator property

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue
https://gravitee.atlassian.net/browse/APIM-4347

## Description

improve openapi-apis by adding required on discriminator property.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-kfkpnqscuy.chromatic.com)

### CODE REVIEW

**Comment by ThibaudAV** [comment]
4.1.x and 4.0.x too :note-sure: wdyt @phiz71 ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml [hunk 1]
```diff
@@ -3221,8 +3221,8 @@ components:
                           example: 2023-05-25T12:40:46.184Z
                       validation:
                           $ref: "#/components/schemas/PlanValidation"
-            required:
-                - definitionVersion
+                  required:
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml:3224]: Are you sure about this ? DefinitionVersion comes from the GenericApi, not from the custom object

**Comment by phiz71** [comment]
> 4.1.x and 4.0.x too :note-sure: wdyt @phiz71 ?

It's not a high priority, but if it's not too complicated to backport, I think it could be interesting.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml (hunk 1 above)
Comment (phiz71) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-apis.yaml:3224]: Forget about it, I just needed to open my 👀

**Review by phiz71 — APPROVED** [review]

**Review by jourdiw — APPROVED** [review]

**Pull request merged**